cmd/ocitools/generate.go

```diff
@@ -0,0 +1,345 @@
+	}
+
+	if context.IsSet("hostname") {
+		g.SetHostname(context.String("hostname"))
+	}
+
+	g.SetPlatformOS(context.String("os"))
+	g.SetPlatformArch(context.String("arch"))
+
+	spec.Annotations = make(map[string]string)
+	if context.IsSet("label") {
+		g.SetAnnotations(context.StringSlice("label"))
+	}
+
+	g.SetRootPath(context.String("rootfs"))
+
+	if context.IsSet("read-only") {
+		g.SetRootReadonly(context.Bool("read-only"))
+	}
+
+	if context.IsSet("uid") {
+		g.SetProcessUID(uint32(context.Int("uid")))
+	}
+
+	if context.IsSet("gid") {
+		g.SetProcessGID(uint32(context.Int("gid")))
+	}
+
+	if context.IsSet("selinux-label") {
+		g.SetProcessSelinuxLabel(context.String("selinux-label"))
+	}
+
+	g.SetProcessCwd(context.String("cwd"))
+
+	if context.IsSet("apparmor") {
+		g.SetProcessApparmorProfile(context.String("apparmor"))
+	}
+
+	if context.IsSet("no-new-privileges") {
+		g.SetProcessNoNewPrivileges(context.Bool("no-new-privileges"))
+	}
+
+	if context.IsSet("tty") {
+		g.SetProcessTerminal(context.Bool("tty"))
+	}
+
+	if context.IsSet("args") {
+		g.SetProcessArgs(context.StringSlice("args"))
+	}
+
+	if context.IsSet("env") {
+		g.SetProcessEnv(context.StringSlice("env"))
+	}
+
+	if context.IsSet("groups") {
+		g.SetProcessAdditionalGids(context.StringSlice("groups"))
+	}
+
+	if context.IsSet("cgroups-path") {
+		g.SetLinuxCgroupsPath(context.String("cgroups-path"))
+	}
+
+	if context.IsSet("mount-label") {
+		g.SetLinuxMountLabel(context.String("mount-label"))
+	}
+
+	if context.IsSet("sysctl") {
+		g.SetLinuxSysctl(context.StringSlice("sysctl"))
+	}
+
+	privileged := false
+	if context.IsSet("privileged") {
+		privileged = context.Bool("privileged")
+	}
+	g.SetupPrivilege(privileged)
+
+	if context.IsSet("cap-add") {
+		addCaps := context.StringSlice("cap-add")
+		if err := g.AddProcessCapabilities(addCaps); err != nil {
+			return err
+		}
+	}
+
+	if context.IsSet("cap-drop") {
+		dropCaps := context.StringSlice("cap-drop")
+		if err := g.DropProcessCapabilities(dropCaps); err != nil {
+			return err
+		}
+	}
+
+	needsNewUser := false
+
+	var uidMaps, gidMaps []string
+
+	if context.IsSet("uidmappings") {
+		uidMaps = context.StringSlice("uidmappings")
+	}
+
+	if context.IsSet("gidmappings") {
+		gidMaps = context.StringSlice("gidmappings")
+	}
+
+	if len(uidMaps) > 0 || len(gidMaps) > 0 {
+		needsNewUser = true
+	}
+
+	nsMaps := map[string]string{}
+	for _, nsName := range generate.Namespaces {
+		if context.IsSet(nsName) {
+			nsMaps[nsName] = context.String(nsName)
+		}
+	}
+	setupLinuxNamespaces(g, needsNewUser, nsMaps)
+
+	if context.IsSet("tmpfs") {
+		tmpfsSlice := context.StringSlice("tmpfs")
+		if err := g.AddTmpfsMounts(tmpfsSlice); err != nil {
+			return err
+		}
+	}
+
+	mountCgroupOption := context.String("mount-cgroups")
+	if err := g.AddCgroupsMounts(mountCgroupOption); err != nil {
+		return err
+	}
+
+	if context.IsSet("bind") {
+		binds := context.StringSlice("bind")
+		if err := g.AddBindMounts(binds); err != nil {
```

Comment:
Is it okay we expect the user to put a single bind mount setting into a slice?
Another I can do is use variadic function.